u2f-host/u2fmisc.c

```diff
@@ -222,37 +222,45 @@ u2fh_sendrecv (u2fh_devs * devs, unsigned index, uint8_t cmd,
     int timeout = HID_TIMEOUT;
     int rc = 0;
 
-    while (rc == 0)
+    do
       {
-	if (debug)
+	timeout = HID_TIMEOUT;
+	rc = 0;
+
+	while (rc == 0)
 	  {
-	    fprintf (stderr, "now trying with timeout %d\n", timeout);
+	    if (debug)
+	      {
+		fprintf (stderr, "now trying with timeout %d\n", timeout);
+	      }
+	    rc = hid_read_timeout (dev->devh, data, len, timeout);
+	    timeout *= 2;
+	    if (timeout > HID_MAX_TIMEOUT)
+	      {
+		rc = -2;
+		break;
+	      }
 	  }
-	rc = hid_read_timeout (dev->devh, data, len, timeout);
-	timeout *= 2;
-	if (timeout > HID_MAX_TIMEOUT)
+	sequence = 0;
+
+	if (debug)
 	  {
-	    rc = -2;
-	    break;
+	    fprintf (stderr, "USB read rc read %d\n", len);
+	    if (rc > 0)
+	      {
+		fprintf (stderr, "USB recv: ");
+		dumpHex (data, 0, rc);
+	      }
 	  }
-      }
-    sequence = 0;
-
-    if (debug)
-      {
-	fprintf (stderr, "USB read rc read %d\n", len);
-	if (rc > 0)
+	if (rc < 0)
 	  {
-	    fprintf (stderr, "USB recv: ");
-	    dumpHex (data, 0, rc);
+	    return U2FH_TRANSPORT_ERROR;
 	  }
+
+	memcpy (&frame, data, HID_RPT_SIZE);
```

Comment:
if we're going to ignore data for this and loop again, should we touch that data at all?

Reply:
It's used in the loop condition, so I think it makes sense.